packages/next/client/head-manager.ts

```diff
@@ -8,6 +8,26 @@ const DOMAttributeNames: Record<string, string> = {
   httpEquiv: 'http-equiv',
 }
 
+function reactComponentToDOM(tag: JSX.Element): HTMLElement[] {
+  let newTags = [] as HTMLElement[]
+  let elTag: JSX.Element
+  try {
+    elTag = tag.type() as JSX.Element
+  } catch (e) {
+    // it's a class, invoke differently
+    elTag = new tag.type().render() as JSX.Element
+  }
+
+  if (elTag.props?.children) {
+    newTags = elTag.props.children.map(
+      (t: JSX.Element): HTMLElement => reactElementToDOM(t)
+    )
+  } else {
+    newTags.push(reactElementToDOM(elTag))
+  }
+  return newTags
+}
+
```

Comment:
So, I don't think it's feasible for Next.js to pass context from the parent tree to custom components in `<Head>`. This means there's an easier way to implement this: just use `ReactDOMServer.renderToString()` and stuff the HTML into the `innerHTML` of a `document.createElement('head')` (we should verify that approach will work in IE11/Edge)